docs/common/craft-parts/reference/plugins/jlink_plugin.rst

```diff
@@ -0,0 +1,99 @@
+.. _craft_parts_jlink_plugin:
+
+JLink plugin
+=============
+
+The JLink plugin can be used for Java projects where you would want to
+deploy Java runtime specific for your application or install a minimal
+Java runtime.
+
+
+Keywords
+--------
+
+This plugin uses the common :ref:`plugin <part-properties-plugin>` keywords as
+well as those for :ref:`sources <part-properties-sources>`.
+
+Additionally, this plugin provides the plugin-specific keywords defined in the
+following sections.
+
+jlink-jars
+~~~~~~~~~~~~~~~~~~
+**Type:** list of strings
+
+List of paths to jar files of your application. If not specified, plugin
+will find all jar files in the staging area.
+
+Dependencies
+------------
+
+The plugin expects OpenJDK to be available on the system and contain
+``jlink`` executable, unless a part named ``jlink-deps`` is defined.
+In this case, the plugin will assume that this part will stage the
+openjdk to be used in the build step.
+
+If the system has multiple OpenJDK installations available, it
+should be selected by setting the ``JAVA_HOME`` environment variable.
+
+.. code-block:: yaml
+
+    parts:
+        runtime:
+            plugin: jlink
+            build-packages:
+                - openjdk-21-jdk
+            build-environment:
+                - JAVA_HOME: /usr/jvm/java-21-openjdk-${CRAFT_ARCH_BUILD_FOR}
```

Comment:
Per our conventions, all the YAML blocks should use two spaces of indentation after the regular text-level indentation. So it should look like this:

```
.. code-block:: yaml

    parts:
      runtime:
        plugin: jlink
    ...
```